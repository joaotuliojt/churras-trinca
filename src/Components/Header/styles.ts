import { styled } from '@styles/stitches'

export const Container = styled('header', {
  height: '12.6rem',
  backgroundImage: 'url("/assets/background.svg")',
  backgroundRepeat: 'repeat',
  padding: '0 4.8rem',
  backgroundColor: '$yellow',
  backgroundSize: 'contain'
})

export const Content = styled('div', {
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-between',
  height: '100%'
})

export const LogoWrap = styled('div', {
  display: 'flex',
  alignItems: 'center',
  gap: '3rem',
  img: {
    width: '5.4rem',
    height: '5.4rem'
  },
  p: {
    fontWeight: 600,
    fontSize: '2rem',
    userSelect: 'none'
  },
  '@media(max-width: 460px)': {
    p: {
      display: 'none'
    }
  }
})

export const Logout = styled('div', {
  a: {
    display: 'flex',
    alignItems: 'center',
    textDecoration: 'none',
    fontWeight: 600,
    gap: '.6rem',
    color: '#000',
    fontSize: '2rem'
  }
})
