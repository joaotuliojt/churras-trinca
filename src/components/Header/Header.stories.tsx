import { ComponentMeta, ComponentStory } from '@storybook/react'
import { BrowserRouter } from 'react-router-dom'
import Header from '.'

export default {
  title: 'Components/Header',
  component: Header,
  decorators: [
    (Story) => (
      <BrowserRouter>
        <Story />
      </BrowserRouter>
    )
  ]
} as ComponentMeta<typeof Header>

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const Template: ComponentStory<typeof Header> = (args) => <Header />

export const Basic = Template.bind({})
