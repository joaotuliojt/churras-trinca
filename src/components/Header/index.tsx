import * as S from './styles'
import { Link } from 'react-router-dom'
import { SignOut } from 'phosphor-react'

const Header = () => (
  <S.Container>
    <S.Content>
      <S.LogoWrap>
        <img src="/assets/trinca.svg" alt="" />
        <p>Agenda de churras</p>
      </S.LogoWrap>
      <S.Logout>
        <Link to={'/'}>
          <SignOut width={20} height={20} />
          Sair
        </Link>
      </S.Logout>
    </S.Content>
  </S.Container>
)

export default Header
