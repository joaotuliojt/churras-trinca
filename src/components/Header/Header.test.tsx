import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'

import Header from '.'

describe('<Header />', () => {
  it('should remove text in mobile', () => {
    const { container, getByText } = render(<Header />, {
      wrapper: BrowserRouter
    })

    console.log(container.style)

    expect(getByText('Agenda de churras')).toBeInTheDocument()
    window.resizeTo(460, 1280)
    //Stitches dont has support to get Styles
    //expect(getByText('Agenda de churras')).toHaveStyle('display: none;')
    expect(container.firstChild).toMatchSnapshot()
  })
})
