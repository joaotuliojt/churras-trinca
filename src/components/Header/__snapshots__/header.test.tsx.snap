// Vitest Snapshot v1

exports[`<Header /> > should remove text in mobile 1`] = `
<header
  class="c-jiJVMe"
>
  <div
    class="c-cYHMdr"
  >
    <div
      class="c-bJztdh"
    >
      <img
        alt=""
        src="/assets/trinca.svg"
      />
      <p>
        Agenda de churras
      </p>
    </div>
    <div
      class="c-cBoDjz"
    >
      <a
        href="/"
      >
        <svg
          fill="currentColor"
          height="20"
          viewBox="0 0 256 256"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <polyline
            fill="none"
            points="174 86 216 128 174 170"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <line
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
            x1="104"
            x2="216"
            y1="128"
            y2="128"
          />
          <path
            d="M104,216H48a8,8,0,0,1-8-8V48a8,8,0,0,1,8-8h56"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
        </svg>
        Sair
      </a>
    </div>
  </div>
</header>
`;
